### DISCUSSION TITLE
Correct format for start and end query parameters filtering

### DISCUSSION THREAD

**Discussion by amzKegor** [weather-gov/api#295] [Q&A]
Regarding to [this documentation](https://api.weather.gov/openapi.json) there is a way to filter alerts between dates using the `start` and `end` query parameters. I tried many formats including the same format that the api uses to return dates and always got a BadRequest with this message `Parameter \"start\" is invalid: This value is not a valid datetime`.
Could you please provide the correct format to use to filter by date?
Thanks

**Comment by netbymatt** [comment]
Hopefully we'll get a formal answer, but I had the same problem and did a lot of experimenting. I found the following:

 - The time is formatted ISO-8601 with several quirks as follows. Keep in mind that all of the following are valid ISO-8601.
 - These formats work
   - `2020-05-14T05:40:08-00:00`
   - `2020-05-14T05:40:08-05:00` Changed offset to -5 hours
   - `2020-05-14T05:40:08Z` Changed time zone to Z (shorthand for UTC)
 - These do not work
   - `2020-05-14T05:40:08+00:00` No positive offsets are accepted
   - `2020-05-14T05:40:08.000Z` Milliseconds are not accepted
   - `20200514T054008Z` Without punctuation is not accepted
   - `2020-05-14` Dates only are not accepted

I also learned the following:

 - The `start` and `end` url parameters appear to be filtering the `sent` field in the responses.
 - The `/alerts` endpoint has data for approximately the last week.

Original Post: https://stackoverflow.com/questions/50473592/api-weather-gov-start-and-end-times/61798862#61798862

**Comment by amzKegor** [comment]
@netbymatt thanks for your answer. Looks like the `start` filter only works if I don't add the `active=true`. When it added, the api returns `BadRequest`. Strange behavior, maybe a bug
